compiler/next/test/resolution/testCanPass.cpp

```diff
@@ -349,18 +366,208 @@ static void test6() {
+                                      ClassType::get(context, basicChild,
+                                                     nullptr, borrowedQ));
+  auto unmanagedParent = QualifiedType(QualifiedType::VAR,
+                                       ClassType::get(context, basicParent,
+                                                      nullptr, unmanaged));
+  auto unmanagedChild = QualifiedType(QualifiedType::VAR,
+                                      ClassType::get(context, basicChild,
+                                                     nullptr, unmanaged));
+  auto unmanagedParentQ = QualifiedType(QualifiedType::VAR,
+                                        ClassType::get(context, basicParent,
+                                                       nullptr, unmanagedQ));
+  auto unmanagedChildQ = QualifiedType(QualifiedType::VAR,
+                                       ClassType::get(context, basicChild,
+                                                      nullptr, unmanagedQ));
+  auto ownedParent = QualifiedType(QualifiedType::VAR,
+                                   ClassType::get(context, basicParent,
+                                                  AnyOwnedType::get(context),
+                                                  owned));
+  auto ownedChild = QualifiedType(QualifiedType::VAR,
+                                  ClassType::get(context, basicChild,
+                                                 AnyOwnedType::get(context),
+                                                 owned));
+  auto ownedParentQ = QualifiedType(QualifiedType::VAR,
+                                    ClassType::get(context, basicParent,
+                                                   AnyOwnedType::get(context),
+                                                   ownedQ));
+  auto ownedChildQ = QualifiedType(QualifiedType::VAR,
+                                   ClassType::get(context, basicChild,
+                                                  AnyOwnedType::get(context),
+                                                  ownedQ));
+
+  CanPassResult r;
+
+  // first, check same type passes
+  r = canPass(borrowedParent, borrowedParent); assert(passesAsIs(r));
+  r = canPass(borrowedChild, borrowedChild); assert(passesAsIs(r));
+  r = canPass(borrowedParentQ, borrowedParentQ); assert(passesAsIs(r));
+  r = canPass(borrowedChildQ, borrowedChildQ); assert(passesAsIs(r));
+  r = canPass(unmanagedParent, unmanagedParent); assert(passesAsIs(r));
+  r = canPass(unmanagedChild, unmanagedChild); assert(passesAsIs(r));
+  r = canPass(unmanagedParentQ, unmanagedParentQ); assert(passesAsIs(r));
+  r = canPass(unmanagedChildQ, unmanagedChildQ); assert(passesAsIs(r));
+  r = canPass(ownedParent, ownedParent); assert(passesAsIs(r));
+  r = canPass(ownedChild, ownedChild); assert(passesAsIs(r));
+  r = canPass(ownedParentQ, ownedParentQ); assert(passesAsIs(r));
+  r = canPass(ownedChildQ, ownedChildQ); assert(passesAsIs(r));
```

Comment:
Consider a macro for doing this and aligning so that it reads like a table?
```
TEST(passesAsIs, borrowedParent, borrowedParent);
...
```

Reply:
Macro seems too much here IMO but I will align it like a table